### PR TITLE
log the plugin loading NoClassDefFoundError

### PR DESCRIPTION
**Pull request by yaozhihang**
Fix the issue
https://github.com/3dcitydb/importer-exporter/issues/285

### CODE REVIEW

**Comment by clausnagel** [comment]
Thanks, @yaozhihang. Couldn't we alternatively just log exceptions for those plugins that cannot be loaded (via something like `addException`) and load those plugins that can be loaded?

File: impexp-core/src/main/java/org/citydb/core/plugin/PluginManager.java [hunk 1]
```diff
@@ -58,7 +59,18 @@ public static synchronized PluginManager getInstance() {
...
         JAXBContext context = createJAXBContext();
         ServiceLoader<Plugin> pluginLoader = ServiceLoader.load(Plugin.class, loader);
-        for (Plugin plugin : pluginLoader) {
+
+        List<Plugin> plugins = new ArrayList<>();
+        try {
+            for (Plugin plugin : pluginLoader) {
+                plugins.add(plugin);
+            }
+        } catch (NoClassDefFoundError e) {
```
Comment (clausnagel) [inline impexp-core/src/main/java/org/citydb/core/plugin/PluginManager.java:68]: I suggest catching `Throwable` here

Comment (yaozhihang) [reply impexp-core/src/main/java/org/citydb/core/plugin/PluginManager.java]: done with [952a7fd](https://github.com/3dcitydb/importer-exporter/pull/287/commits/952a7fdbffa340c1c159b9901d10a09f37eed67e)

**Comment by yaozhihang** [comment]
> 

Hm, i think not possible. The exception is thrown in the method `hasNextService` of the ServiceLoader and no plugin info is returned. You have ideas?

**Comment by clausnagel** [comment]
> > 
> 
> Hm, i think not possible. The exception is thrown in the method `hasNextService` of the ServiceLoader and no plugin info is returned. You have ideas?

Yes, I also tested the iterator interface of the ServiceLoader and unfortunately the error is already thrown in the `hasNext` instead of the `next` method. So, it's not really possible to skip just those plugins with errors and load the others.

I nevertheless propose a small change. I think it would be better to use the exception handling of the `PluginManager` class. This way, the CLI fails on errors (default behaviour), while the error is shown in the console window when using the GUI.

[proposal.patch](https://github.com/3dcitydb/importer-exporter/files/11998989/proposal.patch)

**Comment by yaozhihang** [comment]
> > > 
> > 
> > 
> > Hm, i think not possible. The exception is thrown in the method `hasNextService` of the ServiceLoader and no plugin info is returned. You have ideas?
> 
> I also tested the iterator interface of the ServiceLoader and unfortunately the error is already thrown in the `hasNext` instead of the `next` method. So I can confirm that it's not really possible to skip just those plugins with errors and load the others.
> 
> I nevertheless propose a small change. I think it would be better to use the exception handling of the `PluginManager` class. This way, the CLI fails on errors (default behaviour), while the error is shown in the console window when using the GUI.
> 
> [proposal.patch](https://github.com/3dcitydb/importer-exporter/files/11998989/proposal.patch)

thanks @clausnagel. The Proposal is in this commit [d7ff2c3](https://github.com/3dcitydb/importer-exporter/pull/287/commits/d7ff2c3f74bdc66e2dd748ea75540c872e35b170)

**Review by clausnagel — APPROVED** [review]

**Pull request merged**